include/miral/miral/wayland_extensions.h

```diff
@@ -54,7 +57,25 @@ class WaylandExtensions
 private:
     struct Self;
     std::shared_ptr<Self> self;
+
+    friend auto with_extension(
+        WaylandExtensions const& wayland_extensions,
+        std::string const& name, std::function<std::shared_ptr<void>(wl_display*)> builder) -> WaylandExtensions;
 };
+
+/// Add a bespoke Wayland extension
+/// \remark Since MirAL 2.5
+auto with_extension(
+    WaylandExtensions const& wayland_extensions,
+    std::string const& name, std::function<std::shared_ptr<void>(wl_display*)> builder) -> WaylandExtensions;
+
+template<typename... Extensions>
+auto with_extension(WaylandExtensions const& wayland_extensions,
+                   std::string const& name, std::function<std::shared_ptr<void>(wl_display*)> builder,
+                   Extensions... extensions) -> WaylandExtensions
```

Comment:
Do the variadic template arguments have some advantage over just taking a `std::initializer_list`?

Reply:
Simplicity of use and implementation?